internal/dataplane/translator/translate_utils.go

```diff
@@ -96,20 +97,43 @@ func generateKongServiceFromBackendRefWithName(
 	// check if the service is already known, and if not create it
 	service, ok := rules.ServiceNameToServices[serviceName]
 	if !ok {
+		backendRequestTimeout := DefaultServiceTimeout
+
 		service = kongstate.Service{
 			Service: kong.Service{
 				Name:           kong.String(serviceName),
 				Host:           kong.String(serviceHost),
 				Protocol:       kong.String(protocol),
-				ConnectTimeout: kong.Int(DefaultServiceTimeout),
-				ReadTimeout:    kong.Int(DefaultServiceTimeout),
-				WriteTimeout:   kong.Int(DefaultServiceTimeout),
+				ConnectTimeout: kong.Int(backendRequestTimeout),
+				ReadTimeout:    kong.Int(backendRequestTimeout),
+				WriteTimeout:   kong.Int(backendRequestTimeout),
```

Comment:
Instead of performing the initialization here, can we just leave the one below in rows 132-134? We should put it after the `if ok` condition, but it'd look semantically the same to me, with a slight increase in readability and a decrease of code duplication. Does it make sense?